third_party/wabt/src/interp/jit/jit.cc

```diff
@@ -44,6 +48,80 @@ void JITFunction::call() const {
 
   func.func_entry = func_entry_;
   func.code(context, reinterpret_cast<void*>(start));
+
+  std::vector<int32_t> offsets;
+  int32_t current_offset = 0;
+  int result_alignment;
+  for (ValueType result_type : result_types) {
+    ValueInfo valueInfo = LocationInfo::typeToValueInfo(result_type);
+    switch (valueInfo) {
+      case LocationInfo::kFourByteSize:
+        result_alignment = 4;
+        break;
+      case LocationInfo::kEightByteSize:
+        result_alignment = 8;
+        break;
+      case LocationInfo::kSixteenByteSize:
+        result_alignment = 16;
+        break;
+      default:
+        WABT_UNREACHABLE;
+    }
+    if (current_offset % result_alignment != 0) {
+      current_offset = current_offset - current_offset % result_alignment + result_alignment;
+    }
+    offsets.push_back(current_offset);
+    current_offset += result_alignment;
+  }
+
+  int data_index = 0;
+#define push_result(type)                                           \
+  results.push_back(Value::Make(*(reinterpret_cast<type*>(          \
+      reinterpret_cast<std::byte*>(data) + offsets[data_index])))); \
```

Comment:
`std::byte` -> `u8`. Do you need to type cast `data`?
I still don't like the macro, but at least `#undef` it after use.

Reply:
The type cast of `data` is not absolutely needed, but without it the compiler warns that a `void` pointer was used in arithmetic `[-Wpointer-arith]`